decisions/20240313-try-w3up.md

```diff
@@ -0,0 +1,142 @@
+* anything about nft.storage read and filecoin info APIs. We only need to decide that if/when we have nft.storage uploads going through web3.storage and into filecoin deals. This decision is just to ensure we can get to that point.
+* changing how nft.storage UI does uploads. e.g. it's a non-goal to have nft.storage UI upload things in a different way than it does now, which appears to be handling a POST /uploads/ http request with the file in the request body. (iiuc it looks like request body is a CAR)
+
+## Proposed Solution
+
+### convert nft.storage infra to use w3up-client as a normal customer of w3up
+
+>  convert NFT storage infra to use w3up client to act as a normal customer of w3up and go through normal flow to add the data into w3up. 
+
+https://github.com/w3s-project/project-tracking/issues/7
+
+Rationale
+* "Then all existing filecoin pipeline stuff just happens by default"
+* enables proving that web3.storage's w3-filecoin service can be a long-term replacement of dagcargo for nft.storage filecoin dealmaking
+* lowers cost of adoption of web3.storage by nft.storage by proving out the integration incrementally
+
+#### Architectural Changes
+
+* when nft.storage api handles request POST /upload/ (e.g. from the nft.storage ui), the uploaded file should be stored using up.web3.storage in a space associated with an web3.storage account controlled by an administrator of nft.storage
+    * initially, this will only happen when a feature switch is enabled
+
+#### Data Model Changes
+
+Minimal. Only new configuration.
+
+We'll need to decide which storage space that uploads will be added to.
+To start, we'll just making everything store in a configured storage space, and the app will read that from configuration passed in environment variables, similar to env vars in [add-to-web3](https://github.com/web3-storage/add-to-web3?tab=readme-ov-file#generating-a-secret_key-and-proof).
+
+##### `W3_NFTSTORAGE_SPACE` Environment Variable
+
+configures the web3.storage space that nfts will be stored in.
+
+Example: `did:key:z6MkfbsERaJ7rtJQWMWtYMxNED56bhQMgrNu8CRdUjB5LfRp`
+
+##### `W3_NFTSTORAGE_PRINCIPAL` Environment Variable
+
+configures how nft.storage will authenticate to web3.storage when sending invocations to store things. e.g. it may contain an ed25519 keypair as a CID with raw multihash.
+
+##### `W3_NFTSTORAGE_PROOF` Environment Variable
+
+configures the capabilities that nft.storage has access to when interacting with web3.storage to store nfts. These capabilities will usually be UCAN delegations whose audience is the identifier of `W3_NFTSTORAGE_PRINCIPAL`.
+W3_NFTSTORAGE_PROOF needs to have proof rooted in W3_NFTSTORAGE_SPACE that authorize W3_NFTSTORAGE_PRINCIPAL to store in W3_NFTSTORAGE_SPACE.
+
+##### `NFTSTORAGE_W3S_ALLOW_ACCOUNTS` Environment Variable
```

Comment:
I assume you want to use this for initial testing? If so, that sounds good, but then we can remove it as we want all the things

Reply:
Yep. I added a note to this section to make it more explicit this env var may be removed later once it has served its purpose to control rollout. https://github.com/nftstorage/nft.storage/pull/2518/commits/8481203c5bbd1da8fc6986b1a5832b0bb615a16b